elastic/kibana
# Cancel search queries on timeout regardless of rule type

Resolves https://github.com/elastic/kibana/issues/124805.

In this PR, I'm making the following functions automatically cancel search requests whenever the rule execution is cancelled (via timeout).
- `services.scopedClusterClient.asInternalUser.search`
- `services.scopedClusterClient.asCurrentUser.search`

I am also removing the `services.search` functionality because it is now embedded within the `scopedClusterClient` and no longer requires an explicit set of functions. I moved existing usage towards the appropriate `scopedClusterClient` call.

# To verify

You can use the same steps as outlined in https://github.com/elastic/kibana/pull/122435 to verify the searches are being cancelled.
